components/image-builder-bob/pkg/proxy/auth.go

```diff
@@ -54,17 +54,44 @@ func (a authorizerImpl) Authorize(host string) (user, pass string, err error) {
 	return
 }
 
+func (a MapAuthorizer) AddIfNotExists(other MapAuthorizer) MapAuthorizer {
+	res := make(map[string]authConfig)
+	for k, v := range a {
+		res[k] = v
+	}
+	for k, v := range other {
+		if _, ok := a[k]; ok {
+			continue
+		}
+		res[k] = v
+	}
+	return MapAuthorizer(res)
+}
+
 type Authorizer interface {
 	Authorize(host string) (user, pass string, err error)
 }
 
-func NewAuthorizerFromEnvVar(content string) (auth Authorizer, err error) {
+func NewAuthorizerFromDockerEnvVar(content string) (auth MapAuthorizer, err error) {
 	var res struct {
 		Auths map[string]authConfig `json:"auths"`
 	}
 	err = json.Unmarshal([]byte(content), &res)
 	if err != nil {
 		return
 	}
-	return authorizerImpl(res.Auths), nil
+	return MapAuthorizer(res.Auths), nil
+}
+
+func NewAuthorizerFromEnvVar(content string) (auth MapAuthorizer, err error) {
+	if content == "" {
+		return nil, nil
+	}
+
+	var res map[string]authConfig
+	err = json.Unmarshal([]byte(content), &res)
+	if err != nil {
+		return
```

Comment:
Is there any reason for ignoring this error?
```suggestion
		return nil, err
```

Reply:
it's not ignored. Because `err` is a named return argument, the `return` will just return the value last assigned to `err`

Reply:
@csweichel Oh, I didn't know about it. I just learned, thanks.